src/trace/dma-trace.c

```diff
@@ -218,8 +222,26 @@ int dma_trace_host_buffer(struct dma_trace_data *d,
 }
 #endif
 
+static void dma_trace_buffer_free(struct dma_trace_data *d)
+{
+	struct dma_trace_buf *buffer = &d->dmatb;
+	k_spinlock_key_t key;
+
+	key = k_spin_lock(&d->lock);
+
+	rfree(buffer->addr);
+	memset(buffer, 0, sizeof(*buffer));
```

Comment:
why is this needed? I see, that it is just moved from another location, just wondering if we know why we need this?

Reply:
It is to reset the variables within buffer.